### PR TITLE
BP-830 refactor config providers

### PR DESCRIPTION
**Pull request by florinel-chis**
BP-830 refactor config providers

### CODE REVIEW

File: Model/ConfigProvider/Account.php [hunk 1]
```diff
@@ -235,4 +236,389 @@ protected function getMethodConfigProvider($paymentMethod)
 
         return $this->methodConfigProviderFactory->get($methodCode);
     }
+
+
+    /**
+     * {@inheritdoc}
```
Comment (sandermangel) [inline Model/ConfigProvider/Account.php:242]: I see a lot of @inheritdoc but is there actually documentation available?
I'd love to see maybe also some functional documentation such as what settings this actually relates to. Or what it does.

File: Model/ConfigProvider/Method/Afterpay20.php [hunk 2]
```diff
@@ -80,11 +81,108 @@ public function getConfig()
     public function getPaymentFee($storeId = null)
     {
         $paymentFee = $this->scopeConfig->getValue(
-            self::XPATH_AFTERPAY20_PAYMENT_FEE,
+            static::XPATH_AFTERPAY20_PAYMENT_FEE,
             \Magento\Store\Model\ScopeInterface::SCOPE_STORE,
             $storeId
         );
 
         return $paymentFee ? $paymentFee : false;
```
Comment (sandermangel) [inline Model/ConfigProvider/Method/Afterpay20.php:89]: Not a big fan of mixed return types

File: Model/ConfigProvider/Method/Creditcard.php [hunk 3]
```diff
@@ -124,39 +127,38 @@ class Creditcard extends AbstractConfigProvider
...
-                    }
-                }
+                $sorted_array[$cardName] = $sortedPosition++;
             }
         }
 
-        usort($issuers, function ($cardA, $cardB){
-            return $cardA['sort'] - $cardB['sort'];
-        });
+        $issuers = parent::formatIssuers();
```
Comment (sandermangel) [inline Model/ConfigProvider/Method/Creditcard.php:142]: Can we add some comments to these 20(?) lines on whats going on here? Just to make the mental load a bit lighter. And/or maybe move this to separate methods with descriptive names

File: Model/ConfigProvider/Method/Giftcards.php [hunk 4]
```diff
@@ -71,13 +73,13 @@ public function getConfig()
 
         $paymentFeeLabel = $this->getBuckarooPaymentFeeLabel(\Buckaroo\Magento2\Model\Method\Giftcards::PAYMENT_METHOD_CODE);
 
-        $objectManager = \Magento\Framework\App\ObjectManager::getInstance(); 
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
```
Comment (sandermangel) [inline Model/ConfigProvider/Method/Giftcards.php:76]: Can we get rid of this? Line 76 to 81. Or is this for a different PR?

**Comment by florinel-chis** [comment]
included in refactoring branch, not merging to develop atm.

**Pull request closed without merging**